kubetest2-gke/deployer/build_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import "testing"
+
+func TestNormalizeVersion(t *testing.T) {
+	testCases := []struct {
+		name            string
+		version         string
+		expectedVersion string
+		expectError     bool
+	}{
+		{
+			name:            "empty input",
+			version:         "",
+			expectedVersion: "",
+			expectError:     true,
+		},
+		{
+			name:            "only core version",
+			version:         "1.19.4",
+			expectedVersion: "1.19.4-gke.0",
+			expectError:     false,
+		},
+		{
+			name:            "full version",
+			version:         "1.18.10-gke.601",
+			expectedVersion: "1.18.10-gke.601",
+			expectError:     false,
+		},
+		{
+			name:            "core version, pre-existing suffix",
+			version:         "1.17.12+foobar",
+			expectedVersion: "1.17.12-gke.0+foobar",
+			expectError:     false,
+		},
+		{
+			name:            "full version, pre-existing suffix",
+			version:         "1.16.13-gke.401+qwerty",
+			expectedVersion: "1.16.13-gke.401+qwerty",
+			expectError:     false,
+		},
+		{
+			name:            "full version, multiple pre-existing suffix",
+			version:         "1.16.13-gke.401+qwerty+foobar",
+			expectedVersion: "1.16.13-gke.401+qwerty+foobar",
+			expectError:     false,
+		},
+		{
+			name:            "full version, multiple patch, multiple pre-existing suffix",
+			version:         "1.16.13-gke.401.123+qwerty+foobar",
+			expectedVersion: "1.16.13-gke.401.123+qwerty+foobar",
+			expectError:     false,
+		},
+		{
+			name:            "alpha version",
+			version:         "1.16.13-alpha.123",
+			expectedVersion: "1.16.13-gke.123",
+			expectError:     false,
+		},
+		{
+			name:            "beta version, no minor, pre-existing suffix",
+			version:         "1.20.0-beta+qwe123",
+			expectedVersion: "1.20.0-gke.0+qwe123",
```

Comment:
would it make sense to just insert gke.0 and preserve all the other bits, e.g. `1.20.0-gke.0+beta+qwe123`?

Reply:
makes sense, moved it as a suffix and made it `.99.0` instead. single patch didn't seem to work for me.